### ISSUE TITLE
Java 9 Modules - split package errors

### ISSUE THREAD

**Issue by nullterminated** [orientechnologies/orientdb#8001]
### OS: Arch Linux ARM

Create bare maven project in netbeans. Add orientdb-graphdb to dependencies. Create new modules-info.java. Generated result is:

```
module MyApp {
	requires orientdb.graphdb;
	requires orientdb.server;
	requires orientdb.client;
	requires mail;
	requires activation;
	requires jaxb.impl;
	requires jaxb.api;
	requires stax.api;
	requires orientdb.tools;
	requires orientdb.core;
	requires jna;
	requires jna.platform;
	requires snappy.java;
	requires concurrentlinkedhashmap.lru;
	requires blueprints.core;
	requires jettison;
	requires hppc;
	requires commons.configuration;
	requires commons.lang;
	requires commons.digester;
	requires commons.beanutils;
	requires commons.beanutils.core;
	requires commons.logging;
}
```

And fails to compile with errors:

```
-------------------------------------------------------------
COMPILATION ERROR : 
-------------------------------------------------------------
the unnamed module reads package com.orientechnologies.orient.core.metadata.schema from both orientdb.core and orientdb.client
the unnamed module reads package com.orientechnologies.orient.core.db from both orientdb.core and orientdb.client
the unnamed module reads package com.orientechnologies.orient.core.db.document from both orientdb.core and orientdb.client
the unnamed module reads package com.orientechnologies.orient.server.network.protocol.http from both orientdb.server and orientdb.graphdb
the unnamed module reads package org.apache.commons.collections from both commons.beanutils and commons.beanutils.core
the unnamed module reads package org.apache.commons.beanutils.locale.converters from both commons.beanutils and commons.beanutils.core
the unnamed module reads package org.apache.commons.beanutils.locale from both commons.beanutils and commons.beanutils.core
the unnamed module reads package org.apache.commons.beanutils.converters from both commons.beanutils and commons.beanutils.core
the unnamed module reads package org.apache.commons.beanutils from both commons.beanutils and commons.beanutils.core
module commons.beanutils reads package com.orientechnologies.orient.core.metadata.schema from both orientdb.core and orientdb.client
module commons.beanutils reads package com.orientechnologies.orient.core.db from both orientdb.core and orientdb.client
module commons.beanutils reads package com.orientechnologies.orient.core.db.document from both orientdb.core and orientdb.client
module commons.beanutils reads package com.orientechnologies.orient.server.network.protocol.http from both orientdb.graphdb and orientdb.server
module commons.beanutils reads package org.apache.commons.collections from both commons.beanutils and commons.beanutils.core
module commons.beanutils reads package org.apache.commons.beanutils.locale.converters from both commons.beanutils and commons.beanutils.core
module commons.beanutils reads package org.apache.commons.beanutils.locale from both commons.beanutils and commons.beanutils.core
module commons.beanutils reads package org.apache.commons.beanutils.converters from both commons.beanutils and commons.beanutils.core
module commons.beanutils reads package org.apache.commons.beanutils from both commons.beanutils and commons.beanutils.core
module commons.digester reads package com.orientechnologies.orient.core.metadata.schema from both orientdb.core and orientdb.client
module commons.digester reads package com.orientechnologies.orient.core.db from both orientdb.core and orientdb.client
module commons.digester reads package com.orientechnologies.orient.core.db.document from both orientdb.core and orientdb.client
module commons.digester reads package com.orientechnologies.orient.server.network.protocol.http from both orientdb.graphdb and orientdb.server
module commons.digester reads package org.apache.commons.collections from both commons.beanutils and commons.beanutils.core
module commons.digester reads package org.apache.commons.beanutils.locale.converters from both commons.beanutils and commons.beanutils.core
module commons.digester reads package org.apache.commons.beanutils.locale from both commons.beanutils and commons.beanutils.core
module commons.digester reads package org.apache.commons.beanutils.converters from both commons.beanutils and commons.beanutils.core
module commons.digester reads package org.apache.commons.beanutils from both commons.beanutils and commons.beanutils.core
module commons.lang reads package com.orientechnologies.orient.core.metadata.schema from both orientdb.core and orientdb.client
module commons.lang reads package com.orientechnologies.orient.core.db from both orientdb.core and orientdb.client
module commons.lang reads package com.orientechnologies.orient.core.db.document from both orientdb.core and orientdb.client
module commons.lang reads package com.orientechnologies.orient.server.network.protocol.http from both orientdb.graphdb and orientdb.server
module commons.lang reads package org.apache.commons.collections from both commons.beanutils and commons.beanutils.core
module commons.lang reads package org.apache.commons.beanutils.locale.converters from both commons.beanutils and commons.beanutils.core
module commons.lang reads package org.apache.commons.beanutils.locale from both commons.beanutils and commons.beanutils.core
module commons.lang reads package org.apache.commons.beanutils.converters from both commons.beanutils and commons.beanutils.core
module commons.lang reads package org.apache.commons.beanutils from both commons.beanutils and commons.beanutils.core
module commons.configuration reads package com.orientechnologies.orient.core.metadata.schema from both orientdb.core and orientdb.client
module commons.configuration reads package com.orientechnologies.orient.core.db from both orientdb.core and orientdb.client
module commons.configuration reads package com.orientechnologies.orient.core.db.document from both orientdb.core and orientdb.client
module commons.configuration reads package com.orientechnologies.orient.server.network.protocol.http from both orientdb.graphdb and orientdb.server
module commons.configuration reads package org.apache.commons.collections from both commons.beanutils and commons.beanutils.core
module commons.configuration reads package org.apache.commons.beanutils.locale.converters from both commons.beanutils and commons.beanutils.core
module commons.configuration reads package org.apache.commons.beanutils.locale from both commons.beanutils and commons.beanutils.core
module commons.configuration reads package org.apache.commons.beanutils.converters from both commons.beanutils and commons.beanutils.core
module commons.configuration reads package org.apache.commons.beanutils from both commons.beanutils and commons.beanutils.core
module hppc reads package com.orientechnologies.orient.core.metadata.schema from both orientdb.core and orientdb.client
module hppc reads package com.orientechnologies.orient.core.db from both orientdb.core and orientdb.client
module hppc reads package com.orientechnologies.orient.core.db.document from both orientdb.core and orientdb.client
module hppc reads package com.orientechnologies.orient.server.network.protocol.http from both orientdb.graphdb and orientdb.server
module hppc reads package org.apache.commons.collections from both commons.beanutils and commons.beanutils.core
module hppc reads package org.apache.commons.beanutils.locale.converters from both commons.beanutils and commons.beanutils.core
module hppc reads package org.apache.commons.beanutils.locale from both commons.beanutils and commons.beanutils.core
module hppc reads package org.apache.commons.beanutils.converters from both commons.beanutils and commons.beanutils.core
module hppc reads package org.apache.commons.beanutils from both commons.beanutils and commons.beanutils.core
module jettison reads package com.orientechnologies.orient.core.metadata.schema from both orientdb.core and orientdb.client
module jettison reads package com.orientechnologies.orient.core.db from both orientdb.core and orientdb.client
module jettison reads package com.orientechnologies.orient.core.db.document from both orientdb.core and orientdb.client
module jettison reads package com.orientechnologies.orient.server.network.protocol.http from both orientdb.graphdb and orientdb.server
module jettison reads package org.apache.commons.collections from both commons.beanutils and commons.beanutils.core
module jettison reads package org.apache.commons.beanutils.locale.converters from both commons.beanutils and commons.beanutils.core
module jettison reads package org.apache.commons.beanutils.locale from both commons.beanutils and commons.beanutils.core
module jettison reads package org.apache.commons.beanutils.converters from both commons.beanutils and commons.beanutils.core
module jettison reads package org.apache.commons.beanutils from both commons.beanutils and commons.beanutils.core
module blueprints.core reads package com.orientechnologies.orient.core.metadata.schema from both orientdb.core and orientdb.client
module blueprints.core reads package com.orientechnologies.orient.core.db from both orientdb.core and orientdb.client
module blueprints.core reads package com.orientechnologies.orient.core.db.document from both orientdb.core and orientdb.client
module blueprints.core reads package com.orientechnologies.orient.server.network.protocol.http from both orientdb.graphdb and orientdb.server
module blueprints.core reads package org.apache.commons.collections from both commons.beanutils and commons.beanutils.core
module blueprints.core reads package org.apache.commons.beanutils.locale.converters from both commons.beanutils and commons.beanutils.core
module blueprints.core reads package org.apache.commons.beanutils.locale from both commons.beanutils and commons.beanutils.core
module blueprints.core reads package org.apache.commons.beanutils.converters from both commons.beanutils and commons.beanutils.core
module blueprints.core reads package org.apache.commons.beanutils from both commons.beanutils and commons.beanutils.core
module concurrentlinkedhashmap.lru reads package com.orientechnologies.orient.core.metadata.schema from both orientdb.core and orientdb.client
module concurrentlinkedhashmap.lru reads package com.orientechnologies.orient.core.db from both orientdb.core and orientdb.client
module concurrentlinkedhashmap.lru reads package com.orientechnologies.orient.core.db.document from both orientdb.core and orientdb.client
module concurrentlinkedhashmap.lru reads package com.orientechnologies.orient.server.network.protocol.http from both orientdb.graphdb and orientdb.server
module concurrentlinkedhashmap.lru reads package org.apache.commons.collections from both commons.beanutils and commons.beanutils.core
module concurrentlinkedhashmap.lru reads package org.apache.commons.beanutils.locale.converters from both commons.beanutils and commons.beanutils.core
module concurrentlinkedhashmap.lru reads package org.apache.commons.beanutils.locale from both commons.beanutils and commons.beanutils.core
module concurrentlinkedhashmap.lru reads package org.apache.commons.beanutils.converters from both commons.beanutils and commons.beanutils.core
module concurrentlinkedhashmap.lru reads package org.apache.commons.beanutils from both commons.beanutils and commons.beanutils.core
module snappy.java reads package com.orientechnologies.orient.core.metadata.schema from both orientdb.core and orientdb.client
module snappy.java reads package com.orientechnologies.orient.core.db from both orientdb.core and orientdb.client
module snappy.java reads package com.orientechnologies.orient.core.db.document from both orientdb.core and orientdb.client
module snappy.java reads package com.orientechnologies.orient.server.network.protocol.http from both orientdb.graphdb and orientdb.server
module snappy.java reads package org.apache.commons.collections from both commons.beanutils and commons.beanutils.core
module snappy.java reads package org.apache.commons.beanutils.locale.converters from both commons.beanutils and commons.beanutils.core
module snappy.java reads package org.apache.commons.beanutils.locale from both commons.beanutils and commons.beanutils.core
module snappy.java reads package org.apache.commons.beanutils.converters from both commons.beanutils and commons.beanutils.core
module snappy.java reads package org.apache.commons.beanutils from both commons.beanutils and commons.beanutils.core
module jna.platform reads package com.orientechnologies.orient.core.metadata.schema from both orientdb.core and orientdb.client
module jna.platform reads package com.orientechnologies.orient.core.db from both orientdb.core and orientdb.client
module jna.platform reads package com.orientechnologies.orient.core.db.document from both orientdb.core and orientdb.client
module jna.platform reads package com.orientechnologies.orient.server.network.protocol.http from both orientdb.graphdb and orientdb.server
module jna.platform reads package org.apache.commons.collections from both commons.beanutils and commons.beanutils.core
module jna.platform reads package org.apache.commons.beanutils.locale.converters from both commons.beanutils and commons.beanutils.core
module jna.platform reads package org.apache.commons.beanutils.locale from both commons.beanutils and commons.beanutils.core
module jna.platform reads package org.apache.commons.beanutils.converters from both commons.beanutils and commons.beanutils.core
module jna.platform reads package org.apache.commons.beanutils from both commons.beanutils and commons.beanutils.core
module jna reads package com.orientechnologies.orient.core.metadata.schema from both orientdb.core and orientdb.client
100 errors 
-------------------------------------------------------------
------------------------------------------------------------------------
BUILD FAILURE
------------------------------------------------------------------------
```

**Comment by scotthoye** [comment]
Hello, and thank you to everyone involved with creating OrientDB!  This is my first post (we've been using OrientDB for a few years now, and it has been quite useful!).  We are currently looking into [modularization of our code](https://www.oracle.com/corporate/features/understanding-java-9-modules.html), and this issue (currently marked closed as completed) has become a blocker for us.  Our code depends on **orientdb-core-3.2.16**, **orientdb-client-3.2.16**, **orientdb-server-3.2.16**, and **orientdb-tools-3.2.16**.  Really we only need orientdb-server, but it depends on all of the others (and their transitive dependencies).  The problem is that **orientdb-core** and **orientdb-client** contain [split packages](https://www.logicbig.com/tutorials/core-java-tutorial/modules/split-packages.html), which are not allowed by Java 9+ modules.  Split packages can be detected by the `jdeps` tool, but we actually use this [jsplitpkgscan](https://github.com/AdoptOpenJDK/jsplitpkgscan) tool, as it can better identify split packages. The following split packages are still being identified between orientdb-core-3.2.16 and orientdb-client-3.2.16, which is blocking the ability for us to move forward with Java 9+ modularity.  Can these packages be renamed in a future 3.2.X release, perhaps just in the orientdb-client jar, so that they are unique between the two jars?

Package `com.orientechnologies.orient.core.db` exists in both [orientdb-core-3.2.16](https://github.com/orientechnologies/orientdb/tree/3.2.16/core/src/main/java/com/orientechnologies/orient/core/db) and [orientdb-client-3.2.16](https://github.com/orientechnologies/orientdb/tree/3.2.16/client/src/main/java/com/orientechnologies/orient/core/db)

Package `com.orientechnologies.orient.core.db.document` exists in both [orientdb-core-3.2.16](https://github.com/orientechnologies/orientdb/tree/3.2.16/core/src/main/java/com/orientechnologies/orient/core/db/document) and [orientdb-client-3.2.16](https://github.com/orientechnologies/orientdb/tree/3.2.16/client/src/main/java/com/orientechnologies/orient/core/db/document)

Package `com.orientechnologies.orient.core.metadata.schema` exists in both [orientdb-core-3.2.16](https://github.com/orientechnologies/orientdb/tree/3.2.16/core/src/main/java/com/orientechnologies/orient/core/metadata/schema) and [orientdb-client-3.2.16](https://github.com/orientechnologies/orientdb/tree/3.2.16/client/src/main/java/com/orientechnologies/orient/core/metadata/schema)

Package `com.orientechnologies.orient.core.metadata.security` exists in both [orientdb-core-3.2.16](https://github.com/orientechnologies/orientdb/tree/3.2.16/core/src/main/java/com/orientechnologies/orient/core/metadata/security) and [orientdb-client-3.2.16](https://github.com/orientechnologies/orientdb/tree/3.2.16/client/src/main/java/com/orientechnologies/orient/core/metadata/security)

Hopefully fixing this now would make OrientDB more robust in future Java versions as modularity is used more and more.  Please let me know if I should create a separate ticket instead of reusing this one.  Thanks again!

**Comment by tglman** [comment]
Hi,

Yes it make sense for orientdb to support modularization in long term, will check and try to move the things that can be moved to the right packages, without breaking the APIs.

Bye

**Comment by scotthoye** [comment]
This split package issue appears to be fixed in the latest 3.2.X versions of the orientdb jars.  Thank you!
